react-frontend/src/components/Charts.tsx

```diff
@@ -65,18 +66,42 @@ function Charts({ a_n, b_n, limit, convergesTo, errorData, deltaData, toggleDisp
 
 	const wrapExpression = (input: string, label?: string) => {
 		const expr = label ? label.concat(' = ', input) : input;
+		let parsed, mathy;
 		try {
-			const mathy = parse(expr).toTex();
+			parsed = parse(expr);
+			mathy = parsed.toTex({ parenthesis: 'auto' });
 			return `$$${mathy}$$`;
 		} catch (e) {
-			console.log(`failed to parse ${input}`);
-			return expr;
+			console.error(`failed to parse ${input}`);
+			return `$$${input}$$`;
 		}
 	};
 
+	useEffect(() => {
```

Comment:
this is how we represent the user inputs as a continued fraction at the top of the page - by calculating a and b at each level we wish to show